invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentDetails/components/DocumentAccessRestrictions/state/actions.js

```diff
@@ -0,0 +1,64 @@
+import { IS_LOADING, SUCCESS, HAS_ERROR } from './types';
+import { document as documentApi, patron } from '../../../../../../common/api';
+import {
+  sendErrorNotification,
+  sendSuccessNotification,
+} from '../../../../../../common/components/Notifications';
+import { fetchDocumentDetails } from '../../../state/actions';
+import { ES_DELAY } from '../../../../../../common/config';
+import isEmpty from 'lodash/isEmpty';
+
+export const searchPatrons = queryText => {
+  return async dispatch => {
+    dispatch({
+      type: IS_LOADING,
+    });
+
+    try {
+      const response = await patron.search(queryText);
+      dispatch({
+        type: SUCCESS,
+        payload: response.data,
+      });
+    } catch (error) {
+      dispatch({
+        type: HAS_ERROR,
+        payload: error,
+      });
+      dispatch(sendErrorNotification(error));
+    }
+  };
+};
+
+export const setRestrictionsOnDocument = (pid, accessList) => {
+  const operation = isEmpty(accessList) ? 'remove' : 'add';
+  const ops = [{ op: operation, path: '/_access/read', value: accessList }];
+
+  return async dispatch => {
+    dispatch({
+      type: IS_LOADING,
+    });
+    try {
+      const response = await documentApi.patch(pid, ops);
+      dispatch({
+        type: SUCCESS,
+        payload: response.data,
+      });
+      setTimeout(() => {
+        dispatch(fetchDocumentDetails(pid));
+        dispatch(
+          sendSuccessNotification(
```

Comment:
To be consistent with was decided in previous PRs you need to call the `sendSuccessNotification` before the timeout :)